### PR TITLE
Add ViT-MAE and Refactor ViT(s)

### PR DESCRIPTION
**Pull request by anwai98**
This PR refactors the vision transformers in models in a separate module, as well as implements the vision transformer from MAEs (to make use of their pretrained models).

### CODE REVIEW

**Comment by anwai98** [comment]
@constantinpape This is now good to go. In short, this addresses:
- Fixing UNETR for ViT-SAM
- Adding ViT-MAE to UNETR
- Refactors ViT in `torch_em.model`, at:`vit.py`

File: torch_em/model/unetr.py [hunk 1]
```diff
@@ -215,8 +85,15 @@ def __init__(
     def preprocess(self, x: torch.Tensor) -> torch.Tensor:
         device = "cuda" if torch.cuda.is_available() else "cpu"
 
-        pixel_mean = torch.Tensor([123.675, 116.28, 103.53]).view(-1, 1, 1).to(device)
-        pixel_std = torch.Tensor([58.395, 57.12, 57.375]).view(-1, 1, 1).to(device)
+        if self.use_sam_stats:
+            pixel_mean = torch.Tensor([123.675, 116.28, 103.53]).view(-1, 1, 1).to(device)
+            pixel_std = torch.Tensor([58.395, 57.12, 57.375]).view(-1, 1, 1).to(device)
```
Comment (constantinpape) [inline torch_em/model/unetr.py:90]: I am not sure if this is ever safe to do given that `torch_em` usually pre-normalizes.

File: torch_em/model/unetr.py [hunk 2]
```diff
@@ -3,190 +3,60 @@
...
-            x = blk(x)
-            if i in self.chunks_for_projection:
-                list_from_encoder.append(x)
-
-        x = x.permute(0, 3, 1, 2)
-        list_from_encoder = [e.permute(0, 3, 1, 2) for e in list_from_encoder]
-        return x, list_from_encoder[:3]  # type: ignore
-
-
-def window_partition(x: torch.Tensor, window_size: int) -> Tuple[torch.Tensor, Tuple[int, int]]:
```
Comment (constantinpape) [inline torch_em/model/unetr.py:63]: Just curious, why don't we need those any more?

Comment (anwai98) [reply torch_em/model/unetr.py:63]: Ahha, it is just redundant code. These are parts of ViT from SAM (which I forgot to delete previously hehe)

**Pull request merged**